R/BFF_Demux.R

```diff
@@ -257,129 +257,16 @@ GenerateCellHashCallsBFF <- function(barcodeMatrix, assay = "HTO", min_average_r
-      df <- data.frame(cellbarcode = as.factor(colnames(seuratObj)), method = 'bff_q14', classification = seuratObj$classification.bff_q14, classification.global = seuratObj$classification.global.bff_q14, stringsAsFactors = FALSE)
-      return(df)
-    } else if (calling_paramset == "bff_q15"){
-      SummarizeHashingCalls(seuratObj, label = 'bff_q15', columnSuffix = 'bff_q15', assay = assay, doHeatmap = TRUE)
-      df <- data.frame(cellbarcode = as.factor(colnames(seuratObj)), method = 'bff_q15', classification = seuratObj$classification.bff_q15, classification.global = seuratObj$classification.global.bff_q15, stringsAsFactors = FALSE)
-      return(df)
-    } else if (calling_paramset == "bff_q16"){
-      SummarizeHashingCalls(seuratObj, label = 'bff_q16', columnSuffix = 'bff_q16', assay = assay, doHeatmap = TRUE)
-      df <- data.frame(cellbarcode = as.factor(colnames(seuratObj)), method = 'bff_q16', classification = seuratObj$classification.bff_q16, classification.global = seuratObj$classification.global.bff_q16, stringsAsFactors = FALSE)
-      return(df)
-    } else if (calling_paramset == "bff_q17"){
-      SummarizeHashingCalls(seuratObj, label = 'bff_q17', columnSuffix = 'bff_q17', assay = assay, doHeatmap = TRUE)
-      df <- data.frame(cellbarcode = as.factor(colnames(seuratObj)), method = 'bff_q17', classification = seuratObj$classification.bff_q17, classification.global = seuratObj$classification.global.bff_q17, stringsAsFactors = FALSE)
-      return(df)
-    } else if (calling_paramset == "bff_q18"){
-      SummarizeHashingCalls(seuratObj, label = 'bff_q18', columnSuffix = 'bff_q18', assay = assay, doHeatmap = TRUE)
-      df <- data.frame(cellbarcode = as.factor(colnames(seuratObj)), method = 'bff_q18', classification = seuratObj$classification.bff_q18, classification.global = seuratObj$classification.global.bff_q18, stringsAsFactors = FALSE)
-      return(df)
-    } else if (calling_paramset == "bff_q19"){
-      SummarizeHashingCalls(seuratObj, label = 'bff_q19', columnSuffix = 'bff_q19', assay = assay, doHeatmap = TRUE)
-      df <- data.frame(cellbarcode = as.factor(colnames(seuratObj)), method = 'bff_q19', classification = seuratObj$classification.bff_q19, classification.global = seuratObj$classification.global.bff_q19, stringsAsFactors = FALSE)
-      return(df)
-    } else if (calling_paramset == "bff_q20"){
-      SummarizeHashingCalls(seuratObj, label = 'bff_q20', columnSuffix = 'bff_q20', assay = assay, doHeatmap = TRUE)
-      df <- data.frame(cellbarcode = as.factor(colnames(seuratObj)), method = 'bff_q20', classification = seuratObj$classification.bff_q20, classification.global = seuratObj$classification.global.bff_q20, stringsAsFactors = FALSE)
+    seuratObj <- BFFDemux(seuratObj = seuratObj, assay = assay, simple_threshold = as.logical(simple_threshold), doublet_thresh = doublet_thresh, neg_thresh = neg_thresh, pos_dist = pos_dist, neg_dist = neg_dist)
```

Comment:
why does this need as.logical()? cant we just assume we're given a boolean?

Reply:
Yes.  Thank you for pointing this out.  This was something I tried while debugging and didn't remove.